perceval/backends/gerrit.py

```diff
@@ -0,0 +1,297 @@
+
+        while True:
+            try:
+                raw_items = next(cache_items)
+                if raw_items is None:
+                    break
+                reviews = self._parse_reviews(raw_items)
+            except StopIteration:
+                break
+
+            for item in reviews:
+                yield item
+
+    def fetch(self, from_date=DEFAULT_DATETIME):
+        self._purge_cache_queue()
+
+        self.last_item = None
+        reviews = self._get_reviews()
+
+        while reviews:
+            review = reviews.pop(0)
+            updated = datetime.fromtimestamp(review['lastUpdated'])
+            if updated <= from_date:
+                logging.debug("No more updates for %s" % (self.repository))
+                break
+
+            yield review
+
+            if not reviews:
+                reviews = self._get_reviews()
+
+    def _get_reviews(self):
+        reviews = []
+
+        if self.number_results and self.number_results < self.nreviews:
+            # No more reviews
+            return reviews
+
+        task_init = time()
+        raw_data = self.client.get_items(self.last_item)
+        self._push_cache_queue(raw_data)
+        self._flush_cache_queue()
+        reviews = self._parse_reviews(raw_data)
+        self.last_item = self.client.get_next_item(self.last_item, reviews[-1])
+        logging.info("Received %i reviews in %.2fs" % (len(reviews),
+                                                       time()-task_init))
+        return reviews
+
+    def _parse_reviews(self, raw_data):
+        # Join isolated reviews in JSON in array for parsing
+        items_raw = "[" + raw_data.replace("\n", ",") + "]"
+        items_raw = items_raw.replace(",]", "]")
+        items = json.loads(items_raw)
+        reviews = []
+
+        for item in items:
+            if 'rowCount' in item.keys():
+                # stats data, not a review
+                self.number_results = item['rowCount']
+            elif 'project' in item.keys():
+                reviews.append(item)
+
+        return reviews
+
+
+class GerritClient():
+
+    def __init__(self, repository, user, nreviews):
+        self.gerrit_user = user
+        self.nreviews = nreviews
+        self.repository = repository
+        self.project = None
+        self.version = None
+        self.gerrit_cmd = "ssh -p 29418 %s@%s" % (self.gerrit_user,
+                                                  self.repository)
+        self.gerrit_cmd += " gerrit "
+
+    def _get_version(self):
+
+        if self.version:
+            return self.version
+
+        cmd = self.gerrit_cmd + " version "
+
+        logging.debug("Getting version: %s" % (cmd))
+        raw_data = subprocess.check_output(cmd, shell=True)
+        raw_data = str(raw_data, "UTF-8")
+        logging.debug("Gerrit version: %s" % (raw_data))
+
+        # output: gerrit version 2.10-rc1-988-g333a9dd
+        m = re.match("gerrit version (\d+)\.(\d+).*", raw_data)
```

Comment:
This regular exp pattern should be stored on a class variable. Use the prefix `r''` to tag it as a regular expression.